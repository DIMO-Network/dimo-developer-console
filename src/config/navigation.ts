import {
  HomeIcon,
  IntegrationIcon,
  MonitorHeartIcon,
  SettingsIcon,
  SummarizeIcon,
  SupportAgentIcon,
} from '@/components/Icons';

const APP_DETAILS_REGEX = /^\/app\/details\/[^/]+$/;
const LICENSE_DETAILS_REGEX = /^\/license\/details\/[^/]+$/;
const LICENSED_VEHICLES_REGEX = /^\/license\/vehicles\/[^/]+$/;
export const getPageTitle = (path: string) => {
  const staticPageTitle = pageTitles[path];
  if (staticPageTitle) return staticPageTitle;
  if (APP_DETAILS_REGEX.test(path)) return 'App Details';
  if (LICENSE_DETAILS_REGEX.test(path)) return 'License Details';
  if (LICENSED_VEHICLES_REGEX.test(path)) return 'Licensed Vehicles';
};

const pageTitles: Record<string, string> = {
  '/': 'Home',
  '/app': 'Home',
  '/integrations': 'Integrations',
  '/api-status': 'API Status',
  '/settings': 'Settings',
};

export const mainMenu = [
  {
    label: 'Home',
    icon: HomeIcon,
    iconClassName: 'h-5 w-5',
    link: '/app',
    external: false,
    disabled: false,
  },
  {
    label: 'Integrations',
    icon: IntegrationIcon,
    iconClassName: 'h-5 w-5 fill-white stroke-white stroke-1',
    link: '/integrations',
    external: false,
    disabled: false,
    subMenu: [
      {
        label: 'Webhooks',
        link: '/integrations/webhooks',
        external: false,
      },
    ],
  },
  {
    label: 'Support',
    icon: SupportAgentIcon,
    iconClassName: 'h-5 w-5',
    link: 'https://discord.com/channels/892438668453740634/940719111971946546',
    external: true,
    disabled: false,
  },
  {
    label: 'Documentation',
    icon: SummarizeIcon,
    iconClassName: 'h-5 w-5',
    link: 'https://docs.dimo.zone/developer-platform',
    external: true,
    disabled: false,
  },
  {
    label: 'API Status',
    icon: MonitorHeartIcon,
    iconClassName: 'h-5 w-5',
    link: '/api-status',
    external: false,
    disabled: true,
  },
];

export const bottomMenu = [
  {
    label: 'Settings',
    icon: SettingsIcon,
    iconClassName: 'h-5 w-5 fill-grey-200',
    link: '/settings',
    external: false,
    disabled: false,
  },
];
