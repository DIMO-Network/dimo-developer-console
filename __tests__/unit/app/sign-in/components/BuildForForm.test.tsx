import axios from 'axios';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

import { useSearchParams } from 'next/navigation';
import nock from 'nock';

import SignUpPage from '@/app/sign-up/page';

jest.mock('next/navigation');

const replace = window.location.replace;

axios.defaults.adapter = 'http';

beforeEach(() => {
  Object.defineProperty(window, 'location', {
    value: { replace: jest.fn() },
  });
});

afterEach(() => {
  window.location.replace = replace;
});

describe('SignUpPage', () => {
  it('renders the sign in page at the build for form', () => {
    jest
      .mocked<any>(useSearchParams)
      .mockImplementationOnce(() => ({ get: jest.fn(() => 'build-for') }));

    const { container } = render(<SignUpPage />);

    expect(container).toMatchSnapshot();
  });

  it('renders required action error at build for form', async () => {
    jest
      .mocked<any>(useSearchParams)
      .mockImplementationOnce(() => ({ get: jest.fn(() => 'build-for') }));

    render(<SignUpPage />);

    const continueBtn = screen.getByRole('continue-button');
    fireEvent.click(continueBtn);

    const requiredActionElm = await screen.findByText(
      'Select an option to continue'
    );
    expect(requiredActionElm).toBeInTheDocument();
  });

  it('renders required field error at build for form', async () => {
    jest
      .mocked<any>(useSearchParams)
      .mockImplementationOnce(() => ({ get: jest.fn(() => 'build-for') }));

    render(<SignUpPage />);

    const inputBuildFor = screen.getByRole('build-for-something-else-input');
    fireEvent.click(inputBuildFor);
    const continueBtn = screen.getByRole('continue-button');
    fireEvent.click(continueBtn);

    const requiredActionElm = await screen.findByText('This field is required');
    expect(requiredActionElm).toBeInTheDocument();
  });

  it('renders cleaning field at build for form', async () => {
    jest
      .mocked<any>(useSearchParams)
      .mockImplementationOnce(() => ({ get: jest.fn(() => 'build-for') }));

    render(<SignUpPage />);

    const inputBuildFor = screen.getByRole(
      'build-for-something-else-input'
    ) as HTMLInputElement;
    fireEvent.change(inputBuildFor, { target: { value: 'Test' } });
    const mobileElm = screen.getByText('Mobile app - IOS/Android');
    fireEvent.click(mobileElm);

    expect(inputBuildFor.value).toBe('');
  });

  it('submitting the build for form', async () => {
    jest
      .mocked<any>(useSearchParams)
      .mockImplementationOnce(() => ({ get: jest.fn(() => 'build-for') }));

    nock('http://localhost:3000')
      .persist()
      .intercept('/api/user', 'OPTIONS')
      .reply(200)
      .put('/api/user')
      .reply(200);

    window.location.replace = jest.fn();

    render(<SignUpPage />);

    const mobileElm = screen.getByText('Mobile app - IOS/Android');
    fireEvent.click(mobileElm);

    const continueBtn = screen.getByRole('continue-button');
    fireEvent.click(continueBtn);

    await waitFor(() =>
      expect(window.location.replace).toHaveBeenCalledWith('/sign-up')
    );
  });
});
