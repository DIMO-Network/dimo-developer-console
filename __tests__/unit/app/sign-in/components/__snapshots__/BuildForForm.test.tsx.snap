// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage renders the sign in page at the build for form 1`] = `
<div>
  <main
    class="sign-up"
  >
    <div
      class="sign-up__content"
    >
      <article
        class="sign-up__form"
      >
        <section
          class="sign-up__header"
        >
          <img
            alt="DIMO Logo"
            class="w-44 h-6"
            src="/images/build-on-dimo.png"
          />
          <p>
            What are you building?
          </p>
        </section>
        <form
          class="flex flex-col gap-4 w-full max-w-sm pt-4"
        >
          <div
            class="card flex flex-row card-border justify-between cursor-pointer"
          >
            <p
              class="text-xs font-medium"
            >
              Mobile app - IOS/Android
            </p>
            <svg
              class="w-4 h-5"
              fill="none"
              role="phone-icon"
              viewBox="0 0 15 23"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.5 0.281982H3.5C1.84 0.281982 0.5 1.62198 0.5 3.28198V19.282C0.5 20.942 1.84 22.282 3.5 22.282H11.5C13.16 22.282 14.5 20.942 14.5 19.282V3.28198C14.5 1.62198 13.16 0.281982 11.5 0.281982ZM9 20.282H6C5.72 20.282 5.5 20.062 5.5 19.782C5.5 19.502 5.72 19.282 6 19.282H9C9.28 19.282 9.5 19.502 9.5 19.782C9.5 20.062 9.28 20.282 9 20.282ZM12.5 17.282H2.5V3.28198H12.5V17.282Z"
                fill="white"
              />
              <path
                d="M11.5 0.281982H3.5C1.84 0.281982 0.5 1.62198 0.5 3.28198V19.282C0.5 20.942 1.84 22.282 3.5 22.282H11.5C13.16 22.282 14.5 20.942 14.5 19.282V3.28198C14.5 1.62198 13.16 0.281982 11.5 0.281982ZM9 20.282H6C5.72 20.282 5.5 20.062 5.5 19.782C5.5 19.502 5.72 19.282 6 19.282H9C9.28 19.282 9.5 19.502 9.5 19.782C9.5 20.062 9.28 20.282 9 20.282ZM12.5 17.282H2.5V3.28198H12.5V17.282Z"
                fill="url(#paint0_radial_1649_267)"
                fill-opacity="0.6"
              />
              <defs>
                <radialgradient
                  cx="0"
                  cy="0"
                  gradientTransform="translate(10.2337 9.77984) rotate(171.227) scale(9.84894 70.4556)"
                  gradientUnits="userSpaceOnUse"
                  id="paint0_radial_1649_267"
                  r="1"
                >
                  <stop
                    offset="0.260417"
                    stop-color="#6EE4DA"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="1"
                    stop-color="#B1E3FF"
                  />
                </radialgradient>
              </defs>
            </svg>
          </div>
          <div
            class="card flex flex-row card-border justify-between cursor-pointer"
          >
            <p
              class="text-xs font-medium"
            >
              Web application with single account login
            </p>
            <svg
              class="w-5 h-5"
              fill="none"
              role="computer-icon"
              viewBox="0 0 25 25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_1493_10913)"
              >
                <path
                  d="M20.5 18.282C21.6 18.282 22.49 17.382 22.49 16.282L22.5 6.28198C22.5 5.18198 21.6 4.28198 20.5 4.28198H4.5C3.4 4.28198 2.5 5.18198 2.5 6.28198V16.282C2.5 17.382 3.4 18.282 4.5 18.282H0.5V20.282H24.5V18.282H20.5ZM4.5 6.28198H20.5V16.282H4.5V6.28198Z"
                  fill="white"
                />
                <path
                  d="M20.5 18.282C21.6 18.282 22.49 17.382 22.49 16.282L22.5 6.28198C22.5 5.18198 21.6 4.28198 20.5 4.28198H4.5C3.4 4.28198 2.5 5.18198 2.5 6.28198V16.282C2.5 17.382 3.4 18.282 4.5 18.282H0.5V20.282H24.5V18.282H20.5ZM4.5 6.28198H20.5V16.282H4.5V6.28198Z"
                  fill="url(#paint0_radial_1493_10913)"
                  fill-opacity="0.6"
                />
              </g>
              <defs>
                <radialgradient
                  cx="0"
                  cy="0"
                  gradientTransform="translate(17.1864 11.1895) rotate(176.254) scale(16.7221 51.7362)"
                  gradientUnits="userSpaceOnUse"
                  id="paint0_radial_1493_10913"
                  r="1"
                >
                  <stop
                    offset="0.260417"
                    stop-color="#6EE4DA"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="1"
                    stop-color="#B1E3FF"
                  />
                </radialgradient>
                <clippath
                  id="clip0_1493_10913"
                >
                  <rect
                    fill="white"
                    height="24"
                    transform="translate(0.5 0.281982)"
                    width="24"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="card flex flex-row card-border justify-between cursor-pointer"
          >
            <p
              class="text-xs font-medium"
            >
              Web application with multi-account management
            </p>
            <svg
              class="w-4 h-5"
              fill="none"
              role="car-rental-icon"
              viewBox="0 0 15 22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.5 17.282C5.05228 17.282 5.5 16.8343 5.5 16.282C5.5 15.7297 5.05228 15.282 4.5 15.282C3.94772 15.282 3.5 15.7297 3.5 16.282C3.5 16.8343 3.94772 17.282 4.5 17.282Z"
                fill="white"
              />
              <path
                d="M4.5 17.282C5.05228 17.282 5.5 16.8343 5.5 16.282C5.5 15.7297 5.05228 15.282 4.5 15.282C3.94772 15.282 3.5 15.7297 3.5 16.282C3.5 16.8343 3.94772 17.282 4.5 17.282Z"
                fill="url(#paint0_radial_1649_812)"
                fill-opacity="0.6"
              />
              <path
                d="M10.5 17.282C11.0523 17.282 11.5 16.8343 11.5 16.282C11.5 15.7297 11.0523 15.282 10.5 15.282C9.94771 15.282 9.5 15.7297 9.5 16.282C9.5 16.8343 9.94771 17.282 10.5 17.282Z"
                fill="white"
              />
              <path
                d="M10.5 17.282C11.0523 17.282 11.5 16.8343 11.5 16.282C11.5 15.7297 11.0523 15.282 10.5 15.282C9.94771 15.282 9.5 15.7297 9.5 16.282C9.5 16.8343 9.94771 17.282 10.5 17.282Z"
                fill="url(#paint1_radial_1649_812)"
                fill-opacity="0.6"
              />
              <path
                d="M12.75 9.38198C12.73 9.36198 12.72 9.34198 12.7 9.31198C12.32 8.79198 11.78 8.78198 11.78 8.78198H3.22C3.22 8.78198 2.68 8.79198 2.3 9.32198C2.28 9.34198 2.27 9.36198 2.25 9.38198C2.18 9.49198 2.11 9.62198 2.06 9.78198C1.84 10.442 1.32 12.002 0.5 14.472V20.972C0.5 21.422 0.85 21.782 1.28 21.782H1.72C2.15 21.782 2.5 21.422 2.5 20.972V19.782H12.5V20.972C12.5 21.422 12.84 21.782 13.28 21.782H13.72C14.15 21.782 14.5 21.422 14.5 20.972V14.472C13.68 12.012 13.16 10.442 12.94 9.78198C12.89 9.62198 12.82 9.49198 12.75 9.38198ZM3.83 10.782H11.17L11.4 11.472L11.83 12.782H3.17L3.83 10.782ZM12.5 17.782H2.5V14.782H12.5V17.782Z"
                fill="white"
              />
              <path
                d="M12.75 9.38198C12.73 9.36198 12.72 9.34198 12.7 9.31198C12.32 8.79198 11.78 8.78198 11.78 8.78198H3.22C3.22 8.78198 2.68 8.79198 2.3 9.32198C2.28 9.34198 2.27 9.36198 2.25 9.38198C2.18 9.49198 2.11 9.62198 2.06 9.78198C1.84 10.442 1.32 12.002 0.5 14.472V20.972C0.5 21.422 0.85 21.782 1.28 21.782H1.72C2.15 21.782 2.5 21.422 2.5 20.972V19.782H12.5V20.972C12.5 21.422 12.84 21.782 13.28 21.782H13.72C14.15 21.782 14.5 21.422 14.5 20.972V14.472C13.68 12.012 13.16 10.442 12.94 9.78198C12.89 9.62198 12.82 9.49198 12.75 9.38198ZM3.83 10.782H11.17L11.4 11.472L11.83 12.782H3.17L3.83 10.782ZM12.5 17.782H2.5V14.782H12.5V17.782Z"
                fill="url(#paint2_radial_1649_812)"
                fill-opacity="0.6"
              />
              <path
                d="M6.33 2.78198C5.91 1.61198 4.8 0.781982 3.5 0.781982C1.84 0.781982 0.5 2.12198 0.5 3.78198C0.5 5.43198 1.84 6.78198 3.5 6.78198C4.8 6.78198 5.91 5.94198 6.33 4.78198H11.5V6.78198H13.5V4.78198H14.5V2.78198H6.33ZM3.5 4.78198C2.95 4.78198 2.5 4.33198 2.5 3.78198C2.5 3.23198 2.95 2.78198 3.5 2.78198C4.05 2.78198 4.5 3.23198 4.5 3.78198C4.5 4.33198 4.05 4.78198 3.5 4.78198Z"
                fill="white"
              />
              <path
                d="M6.33 2.78198C5.91 1.61198 4.8 0.781982 3.5 0.781982C1.84 0.781982 0.5 2.12198 0.5 3.78198C0.5 5.43198 1.84 6.78198 3.5 6.78198C4.8 6.78198 5.91 5.94198 6.33 4.78198H11.5V6.78198H13.5V4.78198H14.5V2.78198H6.33ZM3.5 4.78198C2.95 4.78198 2.5 4.33198 2.5 3.78198C2.5 3.23198 2.95 2.78198 3.5 2.78198C4.05 2.78198 4.5 3.23198 4.5 3.78198C4.5 4.33198 4.05 4.78198 3.5 4.78198Z"
                fill="url(#paint3_radial_1649_812)"
                fill-opacity="0.6"
              />
              <defs>
                <radialgradient
                  cx="0"
                  cy="0"
                  gradientTransform="translate(10.2337 9.84812) rotate(171.62) scale(9.83876 67.3227)"
                  gradientUnits="userSpaceOnUse"
                  id="paint0_radial_1649_812"
                  r="1"
                >
                  <stop
                    offset="0.260417"
                    stop-color="#6EE4DA"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="1"
                    stop-color="#B1E3FF"
                  />
                </radialgradient>
                <radialgradient
                  cx="0"
                  cy="0"
                  gradientTransform="translate(10.2337 9.84812) rotate(171.62) scale(9.83876 67.3227)"
                  gradientUnits="userSpaceOnUse"
                  id="paint1_radial_1649_812"
                  r="1"
                >
                  <stop
                    offset="0.260417"
                    stop-color="#6EE4DA"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="1"
                    stop-color="#B1E3FF"
                  />
                </radialgradient>
                <radialgradient
                  cx="0"
                  cy="0"
                  gradientTransform="translate(10.2337 9.84812) rotate(171.62) scale(9.83876 67.3227)"
                  gradientUnits="userSpaceOnUse"
                  id="paint2_radial_1649_812"
                  r="1"
                >
                  <stop
                    offset="0.260417"
                    stop-color="#6EE4DA"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="1"
                    stop-color="#B1E3FF"
                  />
                </radialgradient>
                <radialgradient
                  cx="0"
                  cy="0"
                  gradientTransform="translate(10.2337 9.84812) rotate(171.62) scale(9.83876 67.3227)"
                  gradientUnits="userSpaceOnUse"
                  id="paint3_radial_1649_812"
                  r="1"
                >
                  <stop
                    offset="0.260417"
                    stop-color="#6EE4DA"
                    stop-opacity="0.8"
                  />
                  <stop
                    offset="1"
                    stop-color="#B1E3FF"
                  />
                </radialgradient>
              </defs>
            </svg>
          </div>
          <div
            class="card flex flex-col gap-4 card-border"
          >
            <label
              class="label text-xs text-medium"
              for="buildForText"
            >
              Something else
              <input
                class="text-field"
                name="buildForText"
                placeholder="Enter text..."
                role="build-for-something-else-input"
                type="text"
              />
            </label>
          </div>
          <div
            class="flex flex-col items-center"
          />
          <div
            class="flex flex-col pt-4"
          >
            <button
              class="button primary"
              role="continue-button"
              type="submit"
            >
              Continue
            </button>
          </div>
        </form>
      </article>
    </div>
  </main>
</div>
`;
